### PR TITLE
omfwd: add support for network namespaces

### PR DESCRIPTION
**Pull request by BastianStender**
Network namespaces are part of the linux kernel since v2.6.24. This
enables forwarding output into a configurable network namespace. E.g.,

action(type="omfwd" networknamespace="my_ns" target="192.168.1.23"
port="10514")

Signed-off-by: Bastian Stender <bst@pengutronix.de>

### CODE REVIEW

**Comment by BastianStender** [comment]
Hi,

do you have any idea how to fix this on CentOS 6 and FreeBSD?

I suspect that CentOS 6 has a kernel < 3.0 that does not know setns(). I thought that my [#ifdef](https://github.com/rsyslog/rsyslog/pull/1458/files#diff-c099b5b1da959592a560a0970ebbe645R760) would exclude BSD.

Regards,
Bastian Stender

**Comment by rgerhards** [comment]
how about a configure check for setns()?

File: configure.ac [hunk 1]
```diff
@@ -1316,6 +1316,10 @@ AC_ARG_ENABLE(testbench,
 )
 AM_CONDITIONAL(ENABLE_TESTBENCH, test x$enable_testbench = xyes)
 if test "x$enable_testbench" = "xyes"; then
+	AC_CHECK_PROG(IP, [ip], [yes], [no])
```
Comment (rgerhards) [inline configure.ac]: I think we should make this optional, as only one test needs it. The test should skip out if ip is not found. As it is now, the testbench does not work at all on any platform that does not have "ip".

**Comment by BastianStender** [comment]
setns() and ip availability are now checked in configure.ac

**Comment by rgerhards** [comment]
one final thing: would you be willing to contribute a doc update to https://github.com/rsyslog/rsyslog-doc/blob/master/source/configuration/modules/omfwd.rst ? Otherwise, nobody knows about the new feature. It would be great if you could also add an example use case...

**Comment by BastianStender** [comment]
Sure, will do.

**Comment by BastianStender** [comment]
See https://github.com/rsyslog/rsyslog-doc/pull/321 for the documentation pull request.

**Comment by BastianStender** [comment]
I cannot reproduce the error in the test suite. I guess that has nothing to do with my commit really.

Any comments to the pull request or is anything missing?

**Comment by rgerhards** [comment]
This seems to be a race on Travis, I have restarted the test. Note that, as advised on the mailing list, all Solaris tests currently fail as we are working on getting this up and running.

**Comment by BastianStender** [comment]
The tests now look good. Anything left to do?

**Comment by rgerhards** [comment]
all fine as it looks, but I do not merge every day due to other work. Probably today,.

**Pull request merged**